### PR TITLE
Inital version of the Gateway implementation

### PR DESCRIPTION
**Pull request by ravndal**
Features:
	- Most of the implementaion should be in place. #307
	- Recive sms #307
	- Send SMS via ISendSMS interface. #152

The gateway we used during the codeathon, is no longer avaialble, so it's not possible to test the complete flow.
Working on a solution to enable testing without an acutal gateway.

### CODE REVIEW

**Comment by ravndal** [comment]
NB: The connection string in the code can't really do much harm.

File: Source/Infrastructure/TextMessaging/TextMessage.cs [hunk 1]
```diff
@@ -9,8 +9,9 @@ public class TextMessage
         public string OriginNumber { get; set; }
         public string Message { get; set; }
         public string Keyword { get; set; }
+        public string FullMessage { get; set; }
         public string ReceivedAtGatewayNumber { get; set; }
-        public double Longitude { get; set; }
-        public double Latitude { get; set; }
```
Comment (roarfred) [inline Source/Infrastructure/TextMessaging/TextMessage.cs]: Why are we using string representation for the lat/long values?

Comment (ravndal) [reply Source/Infrastructure/TextMessaging/TextMessage.cs]: We're not, I'll change it  👍

**Comment by karolikl** [comment]
@einari, can you take a look when you have time?

**Pull request merged**